### PR TITLE
mercurial: install zsh completions

### PR DESCRIPTION
**Pull request by dtzWill**
###### Motivation for this change

They're useful, you know :).

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/applications/version-management/mercurial/default.nix [hunk 1]
```diff
@@ -50,7 +50,8 @@ in python2Packages.buildPythonApplication {
       chmod u+x $out/share/cgi-bin/hgweb.cgi
 
       # install bash completion
-      install -D -v contrib/bash_completion $out/share/bash-completion/completions/mercurial
+      install -v -m644 -D contrib/bash_completion $out/share/bash-completion/completions/mercurial
+      install -v -m644 -D contrib/zsh_completion $out/share/zsh/site-functions/_hg
```
Comment (veprbl) [inline pkgs/applications/version-management/mercurial/default.nix]: It is going to the nix store anyway, so, maybe, `-m444`?

File: pkgs/applications/version-management/mercurial/default.nix [hunk 2]
```diff
@@ -50,7 +50,8 @@ in python2Packages.buildPythonApplication {
       chmod u+x $out/share/cgi-bin/hgweb.cgi
 
       # install bash completion
-      install -D -v contrib/bash_completion $out/share/bash-completion/completions/mercurial
+      install -v -m644 -D contrib/bash_completion $out/share/bash-completion/completions/mercurial
```
Comment (hedning) [inline pkgs/applications/version-management/mercurial/default.nix]: Completion doesn't work in bash it seems, we need to copy to [`hg`,  `_hg` or `hg.bash`](https://github.com/scop/bash-completion/blob/master/bash_completion#L2052) for `bash-completions` to lazy load the file. Should probably fix it while we're here :)

File: pkgs/applications/version-management/mercurial/default.nix [hunk 3]
```diff
@@ -50,7 +50,8 @@ in python2Packages.buildPythonApplication {
       chmod u+x $out/share/cgi-bin/hgweb.cgi
 
       # install bash completion
```
Comment (c0bw3b) [inline pkgs/applications/version-management/mercurial/default.nix]: And refresh that comment to `install shell completions` or `install bash/zsh completions`

**Comment by dtzWill** [comment]
Sorry for delay, updated as suggested!

Tested bash completion before/after, confirmed didn't work and now does
^_^.

**Pull request merged**